edgedb-protocol/tests/type_descriptors.rs

```diff
@@ -210,10 +288,114 @@ fn object() -> Result<(), Box<dyn Error>> {
                         cardinality: Some(AtMostOne),
                         name: String::from("title"),
                         type_pos: TypePos(0),
+                        source_type_pos: None,
                     }
                 ]
             })
         ]
     );
     Ok(())
 }
+
+#[test]
+fn object_2_0() -> Result<(), Box<dyn Error>> {
+    use edgedb_protocol::common::Cardinality::*;
+    // SELECT Foo {
+    //   id,
+    //   title,
+    //   [IS Bar].body,
+    // }
+    assert_eq!(
+        decode_2_0(bconcat!(
+        b"\0\0\0 \x03\0\0\0\0\0\0\0\0\0\0\0\0\0\0\x01\x01\0\0\0\x08"
+        b"std::str\x01\0\0\0\0\0!\x03\0\0\0\0\0\0\0\0\0\0\0\0\0\0\x01"
+        b"\0\0\0\0\tstd::uuid\x01\0\0\0\0\0\"\n\xc3\xcc\xa7R\x95\xb7"
+        b"\x11\xef\xb4\x87\x1d\x1b\x9f\xa20\x03\0\0\0\x0cdefault::Foo"
+        b"\x01\0\0\0\"\n\r\xdc\xd7\x1e\x95\xb8\x11\xef\x82M!7\x80\\^4"
+        b"\0\0\0\x0cdefault::Bar\x01\0\0\0^\x01\x1dMg\xe7{\xdd]9\x90\x97"
+        b"O\x82\xfa\xd8\xaf7\0\0\x02\0\x04\0\0\0\x01A\0\0\0\t__tname__"
+        b"\0\0\0\x02\0\0\0\0A\0\0\0\x02id\0\x01\0\x02\0\0\0\0o\0\0\0\x05"
+        b"title\0\0\0\x02\0\0\0\0o\0\0\0\x04body\0\0\0\x03"
+        ))?,
```

Comment:
One thing that appears to be missing in our tests is a truncation test -- ie attempting to decode all start-anchored substrings from (0..1) to (0..n) of this descriptor and ensure that they either decode correctly or return `Underflow`.